### PR TITLE
Allow spaces around search operators

### PR DESCRIPTION
**Pull request by ccschmitz**
## Summary

Allows users to use space characters around filter operators. e.g. `span_name=gorm.Query` can now be `span_name = gorm.Query` and behave the same as before.

**Before**
Screenshot 2024-12-04 at 3 37 34 PM

**After**
Screenshot 2024-12-04 at 3 37 20 PM

## How did you test this change?

Play around with different query formats and ensure the behavior of the query language works as expected (returns the correct results) and that the syntax highlighting in the UI is correct for all queries you can throw at it.

## Are there any deployment considerations?

N/A - all existing queries should still work as expected, just expanding the syntax.

## Does this work require review from our design team?

N/A

### CODE REVIEW

File: backend/parser/parser_test.go [hunk 1]
```diff
@@ -50,6 +50,11 @@ func TestWildcardSearch(t *testing.T) {
 	assert.Equal(t, "SELECT * FROM t WHERE SpanName ILIKE '%asdf%' AND ServiceName ILIKE '%-graph%'", sql)
 }
 
+func TestSpacesInSearch(t *testing.T) {
+	sql, _ := buildSqlForQuery("span_name !=  KafkaWorkersOnStrike")
+	assert.Equal(t, "SELECT * FROM t WHERE NOT (toString(SpanName) = 'KafkaWorkersOnStrike')", sql)
```
Comment (Vadman97) [inline backend/parser/parser_test.go:55]: nice :D

File: frontend/src/components/Search/Parser/listener.ts [hunk 2]
```diff
@@ -110,10 +115,13 @@ export class SearchListener extends SearchGrammarListener {
 	}
 
 	exitKey_val_search_expr = (_ctx: Key_val_search_exprContext) => {
-		this.currentExpression.value = this.currentExpression.text.substring(
-			this.currentExpression.key.length +
-				this.currentExpression.operator.length,
+		// everything after the key, operator, and whitespace
+		const regex = new RegExp(
+			`^${this.currentExpression.key}\\s*${this.currentExpression.operator}\\s*(.*)$`,
```
Comment (Vadman97) [inline frontend/src/components/Search/Parser/listener.ts:120]: not sure what the culprit is, but there seems to be a new flicker in the dropdown after a value is entered
https://www.loom.com/share/d5ef948087244c18b0da8292b6792ede

**Review by Vadman97 — APPROVED** [review]
looks good other than the question in the loom, thanks!

File: frontend/src/components/Search/Parser/listener.ts [hunk 3]
```diff
@@ -128,6 +161,10 @@ export class SearchListener extends SearchGrammarListener {
 		this.expressions.push(this.currentExpression)
 		this.currentExpression = { ...DEFAULT_EXPRESSION }
 	}
+
+	// exitEveryRule = (ctx: ParserRuleContext) => {
+	// 	console.log('::: exitEveryRule', ctx.constructor.name, ctx.getText())
+	// }
```
Comment (ccschmitz) [inline frontend/src/components/Search/Parser/listener.ts:167]: TODO: Cleanup!

File: frontend/src/components/Search/Parser/listener.ts (hunk 2 above)
Comment (ccschmitz) [reply frontend/src/components/Search/Parser/listener.ts]: See https://github.com/highlight/highlight/pull/9681#issuecomment-2532068336 - believe this is resolved!

**Pull request merged**